common/src/main/java/com/wynntils/screens/maps/widgets/IconFilterWidget.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.screens.maps.widgets;
+
+import com.google.common.collect.Lists;
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.screens.maps.IconFilterScreen;
+import com.wynntils.utils.EnumUtils;
+import com.wynntils.utils.colors.CommonColors;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.render.RenderUtils;
+import com.wynntils.utils.render.Texture;
+import java.util.List;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.client.gui.components.AbstractWidget;
+import net.minecraft.client.gui.narration.NarrationElementOutput;
+import net.minecraft.network.chat.Component;
+
+public class IconFilterWidget extends AbstractWidget {
+    private final boolean included;
+    private final float iconRenderX;
+    private final float iconRenderY;
+    private final List<Component> tooltip;
+    private final IconFilterScreen filterScreen;
+    private final Texture icon;
+
+    public IconFilterWidget(
+            int x, int y, int width, int height, Texture icon, IconFilterScreen filterScreen, boolean included) {
+        super(x, y, width, height, Component.literal("Icon Filter Widget"));
+        this.icon = icon;
+        this.filterScreen = filterScreen;
+        this.included = included;
+
+        // Calculate x/y position of the icon to keep it centered
+        iconRenderX = (x + width / 2f) - icon.width() / 2f;
+        iconRenderY = (y + height / 2f) - icon.height() / 2f;
+
+        tooltip = included
+                ? List.of(Component.translatable(
+                        "screens.wynntils.iconFilter.filterExclude.tooltip", EnumUtils.toNiceString(icon)))
+                : List.of(Component.translatable(
+                        "screens.wynntils.iconFilter.filterInclude.tooltip", EnumUtils.toNiceString(icon)));
```

Comment:
this too? or if not then you can also make the transformation (`Lists.transform(tooltip, Component::getVisualOrderText)`) here.

Reply:
Again, i don't see a reason why you make the transformation of the tooltip at render time when you are already doing everything else in the constructor.

Reply:
Everywhere else we call `setTooltipForNextRenderPass` is done this way I don't see why this should be an exception